llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -157,15 +157,29 @@ async def _task(
                     )
                 )
 
-    def send_event(self, message: Event) -> None:
+    def send_event(self, message: Event, step: Optional[str] = None) -> None:
         """Sends an event to a specific step in the workflow.
 
-        Currently we send all the events to all the receivers and we let
-        them discard events they don't want. This should be optimized so
-        that we efficiently send events where we know won't be discarded.
+        If step is None, the event is sent to all the receivers and we let
+        them discard events they don't want.
         """
-        for queue in self._queues.values():
-            queue.put_nowait(message)
+        if step is None:
+            for queue in self._queues.values():
+                queue.put_nowait(message)
+
+        if step not in self._get_steps():
+            raise WorkflowValidationError(f"Step {step} does not exist")
```

Comment:
nit: `send_event` is not called during validation, this should probably be some sort of runtime error